src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/RandomNumberGeneratorImplementation.Windows.cs

```diff
@@ -3,18 +3,27 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Diagnostics;
+using System.Runtime.InteropServices;
 
 namespace System.Security.Cryptography
 {
     partial class RandomNumberGeneratorImplementation
     {
-        private void GetBytes(ref byte pbBuffer, int count)
+        private static void GetBytes(ref byte pbBuffer, int count)
         {
             Debug.Assert(count > 0);
 
             Interop.BCrypt.NTSTATUS status = Interop.BCrypt.BCryptGenRandom(ref pbBuffer, count);
             if (status != Interop.BCrypt.NTSTATUS.STATUS_SUCCESS)
                 throw Interop.BCrypt.CreateCryptographicException(status);
         }
+
+        internal static void FillSpan(Span<byte> data)
+        {
+            if (data.Length > 0)
+            {
+                GetBytes(ref MemoryMarshal.GetReference(data), data.Length);
+            }
+        }
```

Comment:
Isn't FillSpan identical in each implementation?  Can it just be in the shared file, and just be part of the Fill implementation there?

Reply:
It is currently.  I think I was being defensive that at some point one of the platforms could end up needing some sort of state managed... but I guess that ambiguous future can deal with things then.

Reply:
Thanks.